File: app/src/main/java/jp/takke/datastats/OverlayUtil.java
package jp.takke.datastats;

import android.content.Context;
import android.os.Build;
import android.provider.Settings;

public class OverlayUtil {

    public static boolean checkOverlayPermission(Context context) {
        if (Build.VERSION.SDK_INT < 23) {
            return true;
        }
        return Settings.canDrawOverlays(context);
    }
}

File: app/src/main/java/jp/takke/datastats/OverlayUtil.kt
package jp.takke.datastats

import android.content.Context
import android.os.Build
import android.provider.Settings

object OverlayUtil {

    fun checkOverlayPermission(context: Context): Boolean {
        return if (Build.VERSION.SDK_INT < 23) {
            true
        } else Settings.canDrawOverlays(context)
    }
}
